webpack use the legacy assert ( `deepStrictEqual` is not a function )

when use the `power-assert@1.3.1` from `webpack@1.12.14` / `babel-loader@6.2.4`, doesn't have the `deepStrictEqual`, it seems old `assert` is used.

```js
import assert from 'assert';
import moduleName from '../src';

describe('moduleName', () => {
  it('spec1', () => {
    assert.deepEqual(moduleName(), { foo: null });// pass
    assert.deepStrictEqual(moduleName(), { foo: null });// error
  });
});
```

simulation environment: https://github.com/59naga/boilerplate-browser-babel

```bash
git clone https://github.com/59naga/boilerplate-browser-babel
cd boilerplate-browser-babel
git checkout 4804fd5432c626268ca2f148bc58e15bed7c0ec5

npm install
node -v && npm -v
# v5.10.0
# 3.8.3
npm test
# TypeError: _powerAssert2.default.deepStrictEqual is not a function
#        at Context.<anonymous> (/Users/59naga/Downloads/boilerplate-browser-babel/test/index.js:71:28 <- webpack:///test/index.js:10:11)
```

one workaround,  frozen the version of `assert` in `4.1.2` using the [core-assert](https://github.com/59naga/power-assert/commit/1046e66da2a2c5be16988620ee68b43613450d33).

```bash
npm install 59naga/power-assert
npm test
# ...
# AssertionError:   # test/index.js:11
#
#   assert.deepStrictEqual(moduleName(), { foo: null })
#                          |                 |
#                          |                 Object{foo:null}
#                          Object{foo:undefined}
```

other than this, would you have any good solution?

> 英語に自信がないので、翻訳前の日本語も残しておきます。
> `webpack`/`babel-loader`から`power-assert`を使用すると、`deepStrictEqual`を持たない、古い`assert`が使用されるようです。
> 解決方法の一つとして、[core-assert](https://github.com/59naga/power-assert/commit/1046e66da2a2c5be16988620ee68b43613450d33)に依存し、`assert`のバージョンを`4.1.2`に固定する方法があります。
> これ以外で、何か良い解決方法はご存知でしょうか？

--- a/index.js
+++ b/index.js
@@ -10,6 +10,7 @@
 'use strict';
 
 var baseAssert = require('assert');
+var _deepEqual = require('universal-deep-strict-equal');
 var empower = require('empower');
 var formatter = require('power-assert-formatter');
 var extend = require('xtend');
@@ -18,6 +19,21 @@ var empowerOptions = {
     modifyMessageOnRethrow: true,
     saveContextOnRethrow: true
 };
+
+if (typeof baseAssert.deepStrictEqual !== 'function') {
+    baseAssert.deepStrictEqual = function deepStrictEqual (actual, expected, message) {
+        if (!_deepEqual(actual, expected, true)) {
+            baseAssert.fail(actual, expected, message, 'deepStrictEqual', deepStrictEqual);
+        }
+    };
+}
+if (typeof baseAssert.notDeepStrictEqual !== 'function') {
+    baseAssert.notDeepStrictEqual = function notDeepStrictEqual (actual, expected, message) {
+        if (_deepEqual(actual, expected, true)) {
+            baseAssert.fail(actual, expected, message, 'notDeepStrictEqual', notDeepStrictEqual);
+        }
+    };
+}
 
 function customize (customOptions) {
     var options = customOptions || {};

--- a/package.json
+++ b/package.json
@@ -48,6 +48,7 @@
     "define-properties": "^1.1.2",
     "empower": "^1.1.0",
     "power-assert-formatter": "^1.3.1",
+    "universal-deep-strict-equal": "^1.1.0",
     "xtend": "^4.0.0"
   },
   "devDependencies": {

--- a/test/tobe_instrumented/assertion.js
+++ b/test/tobe_instrumented/assertion.js
@@ -860,6 +860,18 @@ describe('power-assert message', function () {
     });
 
 
+    it('deepStrictEqual', function () {
+        this.expectPowerAssertMessage(function () {
+            assert.deepStrictEqual({a:1}, {a:'1'});
+        },[
+            '  assert.deepStrictEqual({ a: 1 }, { a: \'1\' })',
+            '                         |         |          ',
+            '                         |         Object{a:\"1\"}',
+            '                         Object{a:1}          '
+        ]);
+    });
+
+
     it('ObjectExpression with Literals', function () {
         this.expectPowerAssertMessage(function () {
             assert.deepEqual({ foo: 0 }, { foo: 1 });

